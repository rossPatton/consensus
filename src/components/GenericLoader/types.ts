export type tProps = {
  isLoading: boolean,
  render: Function;
}
