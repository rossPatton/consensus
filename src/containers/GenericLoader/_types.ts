export type tProps = {
  isLoading: boolean,
  showLoader?: boolean,
  render: Function;
};
