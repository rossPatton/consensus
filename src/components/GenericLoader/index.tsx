import React, { memo } from 'react';

import { tProps } from './_types';

// TODO potentially remove that we are using dynamic imports + loadable
const GenericLoader = memo((props: tProps) => {
  if (props.isLoading) return <></>;
  return props.render();
});

export default GenericLoader;
