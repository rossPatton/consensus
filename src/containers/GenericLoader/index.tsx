import React from 'react';

import { tProps } from './_types';

const GenericLoader = (props: tProps) => {
  const {isLoading, showLoader = true} = props;

  if (isLoading) {
    if (showLoader) {
      return (
        <div className="p-3 w-full flex justify-center">
          <div className="loading">
            <div />
            <div />
            <div />
            <div />
          </div>
        </div>
      );
    }

    return null;
  }

  return props.render();
};

export default GenericLoader;
