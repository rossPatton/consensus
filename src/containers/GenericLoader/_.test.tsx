import React from 'react';
import render from 'react-test-renderer';

import GenericLoader from '.';

describe('containers/GenericLoader', () => {
  it('renders loader', () => {
    const component = render.create((
      <GenericLoader
        isLoading
        render={() => <div />}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty div', () => {
    const component = render.create((
      <GenericLoader
        isLoading={false}
        render={() => <div />}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dont render load screen, just null', () => {
    const component = render.create((
      <GenericLoader
        isLoading
        showLoader={false}
        render={() => <div />}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
