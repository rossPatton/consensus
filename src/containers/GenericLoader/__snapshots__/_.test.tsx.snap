// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/GenericLoader dont render load screen, just null 1`] = `null`;

exports[`containers/GenericLoader renders empty div 1`] = `<div />`;

exports[`containers/GenericLoader renders loader 1`] = `
<div
  className="p-3 w-full flex justify-center"
>
  <div
    className="loading"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
